src/components/data_display/helpers.ts

```diff
@@ -41,44 +40,40 @@ export const getPermissionNamesFromJWT = async (request: any): Promise<JWTData>
 
 export const buildAllColumnDefinitions = async ({
   permissionNames,
-  tableName,
+  dataViewCode,
   type,
   userId,
   orgId,
 }: {
   permissionNames: string[]
-  tableName: string
+  dataViewCode: string
   type: 'TABLE' | 'DETAIL'
   userId: number
   orgId: number | undefined
 }): Promise<ColumnDetailOutput> => {
   // Look up allowed Data views
-  const dataTables = (await DBConnect.getAllTableNames()).map((tableName) => camelCase(tableName))
-
-  const matchingTableName = dataTables.find(
-    (table) => table === dataTablePrefix + capitaliseFirstLetter(tableName) || table === tableName
-  )
-
-  if (!matchingTableName) throw new Error(`Invalid table name: ${tableName}`)
-
-  const dataViews = (await DBConnect.getAllowedDataViews(permissionNames, tableName))
+  const dataViews = (await DBConnect.getAllowedDataViews(permissionNames, dataViewCode))
     .map((dataView) => objectKeysToCamelCase(dataView))
     .sort((a, b) => b.priority - a.priority) as DataView[]
 
-  if (dataViews.length === 0) throw new Error(`No views available for table "${tableName}"`)
+  if (dataViews.length === 0) throw new Error(`No matching data views: "${dataViewCode}"`)
+
+  const dataView = dataViews[0]
+
+  const { tableName, title, code } = dataView
 
-  const { title, code } = dataViews[0]
+  const tableNameProper = camelCase(getValidTableName(tableName))
 
   // Generate graphQL filter object
-  const gqlFilters = getFilters(dataViews, userId, orgId)
+  const gqlFilters = getFilters(dataView, userId, orgId)
 
   // Only for details view
-  const headerColumnName = dataViews[0]?.detailViewHeaderColumn ?? ''
-  const showLinkedApplications = dataViews[0].showLinkedApplications
+  const headerColumnName = dataView.detailViewHeaderColumn ?? ''
+  const showLinkedApplications = dataView.showLinkedApplications
```

Comment:
Downstream stuff gets simpler now that we only deal with a single data view per request.